runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindowingWindmillReader.java

```diff
@@ -116,13 +117,19 @@
     final WorkItem workItem = context.getWork();
     KeyedWorkItem<K, T> keyedWorkItem =
         new WindmillKeyedWorkItem<>(key, workItem, windowCoder, windowsCoder, valueCoder);
+    final boolean isEmptyWorkItem =
```

Comment:
It might be easier to understand the code if we return a new `NativeReaderIterator` that returns false for start/advance and throws NoSuchElementException instead of modifying the existing one.
```
if (isEmptyWorkItem) {
  return new NativeReaderIterator() { ... }
}
...
```